worlds/inscryption/Rules.py

```diff
@@ -0,0 +1,189 @@
+            "Act 3 - Boss Photographer": self.has_inspectometer_battery,
+            "Act 3 - Boss Archivist": self.has_battery_and_quill,
+            "Act 3 - Boss Unfinished": self.has_gems_and_battery,
+            "Act 3 - Boss G0lly": self.has_gems_and_battery,
+            "Act 3 - Extra Battery": self.has_inspectometer_battery,  # Hard to miss but soft lock still possible.
+            "Act 3 - Nano Armor Generator": self.has_gems_and_battery,  # Costs money, so can need multiple battles.
+            "Act 3 - Shop Holo Pelt": self.has_gems_and_battery,  # Costs money, so can need multiple battles.
+            "Act 3 - Middle Holo Pelt": self.has_inspectometer_battery,  # Can be reached without but possible soft lock
+            "Act 3 - Forest Holo Pelt": self.has_inspectometer_battery,
+            "Act 3 - Crypt Holo Pelt": self.has_inspectometer_battery,
+            "Act 3 - Tower Holo Pelt": self.has_gems_and_battery,
+            "Act 3 - Trader 1": self.has_pelts(1),
+            "Act 3 - Trader 2": self.has_pelts(2),
+            "Act 3 - Trader 3": self.has_pelts(3),
+            "Act 3 - Trader 4": self.has_pelts(4),
+            "Act 3 - Trader 5": self.has_pelts(5),
+            "Act 3 - Goobert's Painting": self.has_gems_and_battery,
+            "Act 3 - The Great Transcendence": self.has_transcendence_requirements,
+            "Act 3 - Boss Mycologists": self.has_mycologists_boss_requirements,
+            "Act 3 - Bone Lord Room": self.has_bone_lord_room_requirements,
+            "Act 3 - Luke's File Entry 1": self.has_battery_and_quill,
+            "Act 3 - Luke's File Entry 2": self.has_battery_and_quill,
+            "Act 3 - Luke's File Entry 3": self.has_battery_and_quill,
+            "Act 3 - Luke's File Entry 4": self.has_transcendence_requirements,
+            "Act 3 - Well": self.has_inspectometer_battery,
+            "Act 3 - Gems Drone": self.has_inspectometer_battery,
+            "Act 3 - Clock": self.has_gems_and_battery,  # Can be brute-forced, but the solution needs those items.
+        }
+        self.region_rules = {
+            "Act 2": self.has_act2_requirements,
+            "Act 3": self.has_act3_requirements,
+            "Epilogue": self.has_epilogue_requirements
+        }
+
+    def has_wardrobe_key(self, state: CollectionState) -> bool:
+        return state.has("Wardrobe Key", self.player)
+
+    def has_caged_wolf(self, state: CollectionState) -> bool:
+        return state.has("Caged Wolf Card", self.player)
+
+    def has_dagger(self, state: CollectionState) -> bool:
+        return state.has("Dagger", self.player)
+
+    def has_magnificus_eye(self, state: CollectionState) -> bool:
+        return state.has("Magnificus Eye", self.player)
+
+    def has_useful_act1_items(self, state: CollectionState) -> bool:
+        return state.has("Oil Painting's Clover Plant", self.player) and state.has("Squirrel Totem Head", self.player)
```

Comment:
This can use `has_all`
```suggestion
        return state.has_all(("Oil Painting's Clover Plant", "Squirrel Totem Head"), self.player)
```